lib/IRGen/GenDecl.cpp

```diff
@@ -1172,9 +1172,9 @@ void IRGenModule::emitGlobalLists() {
     if (IRGen.Opts.EmitGenericRODatas) {
       emitGlobalList(
           *this, GenericRODatas, "generic_ro_datas",
-          GetObjCSectionName("__swift_rodatas", "regular"),
+          GetObjCSectionName("__objc_clsrolist", "regular"),
           llvm::GlobalValue::InternalLinkage, Int8PtrTy, /*isConstant*/ false,
-          /*asContiguousArray*/ true, /*canBeStrippedByLinker*/ true);
+          /*asContiguousArray*/ true, /*canBeStrippedByLinker*/ false);
```

Comment:
> /*canBeStrippedByLinker*/ false

Is this about the symbol name or dead code stripping?

Reply:
This boolean decides whether llvm.used or llvm.compiler.used is used.

This will influence whether llvm's LTO can assume that the variable is unused if there are no other uses (that would happen with `llvm.compiler.used` i.e /*canBeStrippedByLinker*/ true) but will also cause the emission of `.no_dead_strip  _generic_ro_datas`.

As in:

```
        .section        __LD,__objc_clsrolist
        .p2align        3, 0x0
_generic_ro_datas:
        .quad   ___unnamed_1+40
        .quad   ___unnamed_1+112
```
I assumed that is okay since the linker will know it is to remove that section at the point it uses it.

Reply:
ah, right we need this for LTO. It means older linkers that don't know about the section won't be able to dead strip it. That's maybe fine though? I assume this won't be too big, and it's only a problem when building with new Swift compiler but using an older linker.

Reply:
There will be one entry per generic class definition. Yes, I assume this to be a marginal problem that if it turns out to be a problem for an individual project that uses a new compiler with an old linker can turn this off via compiler flags.